tests/test_parser.py

```diff
@@ -84,44 +85,65 @@ def test_constructor_cache(self, tmpdir):
-                          " # comment\n",
-                          " label  foo  \\\n",  # extra ws
-                          "    # comment\n",    # should be ignored
-                          "    bar  \n",        # extra ws, continuation line
+        dfparser.lines = ["# comment\n",                # single-line comment
+                          " From  \\\n",                # mixed-case
+                          "   base\n",                  # extra ws, continuation line
+                          " #    another   comment\n",  # extra ws
+                          " label  foo  \\\n",          # extra ws
+                          "# multi-line \n",            # multi-line comment, interrupt instruction
+                          "# ...comment\n",
+                          "    bar  \n",                # extra ws, instruction continuation
                           "USER  {0}\n".format(NON_ASCII),
-                          "# comment \\\n",     # extra ws
-                          "# comment \\ \n",    # extra ws with a space
-                          "# comment \\\\ \n",  # two backslashes
+                          "# comment \\\n",             # extra ws
+                          "# with \\ \n",               # extra ws with a space
+                          "# backslashes \\\\ \n",      # two backslashes
                           "RUN command1\n",
                           "RUN command2 && \\\n",
-                          "    # comment\n",
                           "    command3\n"]
 
-        assert dfparser.structure == [{'instruction': 'FROM',
-                                       'startline': 1,  # 0-based
+        assert dfparser.structure == [
+                                      {'instruction': COMMENT_INSTRUCTION,
+                                       'startline': 0,
+                                       'endline': 0,
+                                       'content': '# comment\n',
+                                       'value': 'comment'},
+                                      {'instruction': 'FROM',
+                                       'startline': 1,
                                        'endline': 2,
                                        'content': ' From  \\\n   base\n',
                                        'value': 'base'},
+                                      {'instruction': COMMENT_INSTRUCTION,
+                                       'startline': 3,
+                                       'endline': 3,
+                                       'content': ' #    another   comment\n',
+                                       'value': 'another   comment'},
+                                      {'instruction': COMMENT_INSTRUCTION,
+                                       'startline': 5,
+                                       'endline': 6,
+                                       'content': '# multi-line \n# ...comment\n',
```

Comment:
Docker has NO multi-line comments. Why we should handle comments like multi-line in parser? I vote for having all comments as single line, there is no format support for multiline comments in Dockerfile

Reply:
Reasoning behind this is that you usually consider several contiguous lines of comments as one comment block. If we remove the multi-line comments, then imagine a very long documentation block - say 20 lines of comments - will be parsed as 20 separate "instruction" dictionaries, which feels a bit weird. Though I can see that supporting multi-lines comment is weird too ... It's your call really I don't care. Let me know and I'll change this.

Reply:
I'd stay with oneline comments only but let's wait for other opinions here

Reply:
Ok since there are no other opinions, I guess I'll just remove the multi-line comments